src/stores/messages/MessagesStore.ts

```diff
@@ -128,6 +135,88 @@ export default class MessagesStore {
 		return [timestampTo - 15 * 1000, timestampTo + 15 * 1000];
 	}
 
+	public includes(message: EventMessage) {
+		return (
+			this.exportMessages.filter(exportMessage => exportMessage.messageId === message.messageId)
+				.length > 0
+		);
+	}
+
+	@action
+	public addMessageToExport(message: EventMessage) {
+		if (!this.isExport) return;
+		if (this.includes(message)) {
+			this.exportMessages = this.exportMessages.filter(
+				exportMessage => exportMessage.messageId !== message.messageId,
+			);
+		} else {
+			this.exportMessages.push(message);
+		}
+	}
+
+	@action
+	public enableExport() {
+		this.isExport = true;
+		this.exportMessages = [];
+	}
+
+	@action
+	public disableExport() {
+		this.isExport = false;
+		this.exportMessages = [];
+	}
+
+	private convertMessage(messageToConvert: EventMessage, messageViewType: MessageViewType) {
+		let content: string;
+
+		const jsonToCopy = messageToConvert.body;
+
+		switch (messageViewType) {
+			case MessageViewType.ASCII:
+				content = messageToConvert.bodyBase64 ? atob(messageToConvert.bodyBase64) : '';
+				break;
+			case MessageViewType.BINARY:
+				content = messageToConvert.bodyBase64
+					? getAllRawContent(decodeBase64RawContent(messageToConvert.bodyBase64))
+					: '';
+				break;
+			case MessageViewType.FORMATTED:
+				content = jsonToCopy ? JSON.stringify(jsonToCopy, null, 4) : '';
+				break;
+			case MessageViewType.JSON:
+				content = jsonToCopy ? JSON.stringify(jsonToCopy) : '';
+				break;
+			default:
+				content = '';
+		}
+
+		return content;
+	}
+
+	@action
+	public endExport(messageViewType: MessageViewType) {
+		this.isExport = false;
+		if (this.exportMessages.length === 0) return;
+		const exportUrl = URL.createObjectURL(
```

Comment:
there is downloadTxtFile function which currently isn't used anywhere, you can modify it and move all file creating-downloading logic there